components/lists/Friends.tsx

```diff
@@ -1,18 +1,43 @@
-import { StyleSheet, View } from "react-native"
+import React, { useState, useEffect } from "react"
+import { View, StyleSheet } from "react-native"
 import CustomText from "@/components/ui/CustomText"
 import FriendCard from "@/components/Friend"
-import { useState } from "react"
-import { availableFriends } from "@/data/users"
+import api from "@/service"
+import { Friend } from "@/types"
+import { useStatus } from "@/contexts/StatusContext"
 
 export default function FriendsList() {
-  const [friendsList, setFriendsList] = useState(availableFriends)
-  const updateFriendsList = (userId: string) => {
-    setFriendsList((prevList) =>
-      prevList.map((user) =>
-        user.id === userId ? { ...user, selected: !user.selected } : user,
-      ),
-    )
+  const { friends, setFriends } = useStatus()
+  const [friendsList, setFriendsList] = useState<Friend[]>([])
+
+  const fetchFriends = async () => {
+    try {
+      const response = await api.get("/friends")
+      const allFriends = response.data.map((friend: any) => ({
+        id: friend.id,
+        name: friend.name,
+        email: friend.email,
+        imageUrl: friend.profile || "",
+        selected: friends.includes(friend.id),
+      }))
+
+      setFriendsList(allFriends)
+    } catch (error) {
+      console.error("Error fetching friends", error)
+    }
+  }
+
+  useEffect(() => {
+    fetchFriends()
+  }, [friends])
```

Comment:
this dependency array value is not making any difference, add `fetchFriends` function in a dependency array instead and wrap the function in a `useCallback`